### PR TITLE
Migration to firestore

### PR DESCRIPTION
**Pull request by PriscilaAlfaro**

### CODE REVIEW

**Review by wirob — CHANGES_REQUESTED** [review]

File: src/controllers/answers.js [hunk 1]
```diff
@@ -26,22 +23,12 @@ router.get('/', async (req, res) => {
...
-      (aProduct) => aProduct._id.toString() === req.params.productId
-    );
-    if (!product) {
-      return res.status(404).json({ message: 'product id does not exist' });
-    }
-
+    const answers = await getAllAnswersForAProduct(req.params.userId, req.params.productId);
     // info for header
-    const totalVotes = product.answers.length;
+    const totalVotes = answers.length;
```
[the remark this answers is not in the document]
Comment (PriscilaAlfaro) [reply src/controllers/answers.js:28]: getAllAnswersForAProduct return an array with all the answers for a specific product. This is used to populate the charts.

File: src/controllers/answers.js [hunk 2]
```diff
@@ -50,14 +37,14 @@ router.get('/formatted', async (req, res) => {
     });
 
     // info for opinions  - flow page 11
-    const whatSomeThink = product.answers.map((answer) => {
+    const whatSomeThink = answers.map((answer) => {
       const receivName = answer.receiverName;
       const receivMessage = answer.receiverMessage;
```
[the remark this answers is not in the document]
Comment (PriscilaAlfaro) [reply src/controllers/answers.js:42]: Understood.

File: src/controllers/products.js [hunk 3]
```diff
@@ -38,12 +42,8 @@ router.param('productId', async (req, res, next) => {
...
-    const user = await User.findById(req.params.userId);
-
-    if (!user) {
-      res.status(404).json({ message: 'user id does not exist' });
-    }
-    return res.json(user.products);
+    const products = await getAllProductsOfAUser(req.params.userId);
+    return res.json(products);
   } catch (err) {
     return res.status(500).send();
```
[the remark this answers is not in the document]
Comment (PriscilaAlfaro) [reply src/controllers/products.js:48]: Understood.

**Review by kosmolet — APPROVED** [review]

File: src/controllers/questions.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 
 const router = express.Router();
 
 router.get('/', async (req, res) => {
   try {
-    const allQuestions = await Question.find();
+    const allQuestions = await getAllQuestions();
     return res.json(allQuestions);
   } catch (err) {
     return res.status(500).send();
```
Comment (wirob) [inline src/controllers/questions.js:15]: Missing the error to be sent to the client here.

**Review by wirob — CHANGES_REQUESTED** [review]

File: src/controllers/questions.js (hunk 4 above)
Comment (wirob) [inline src/controllers/questions.js:15]:
```suggestion
    return res.status(500).send(err);
```

**Review by wirob — CHANGES_REQUESTED** [review]

File: src/controllers/questions.js (hunk 4 above)
Comment (PriscilaAlfaro) [reply src/controllers/questions.js]: Understood 🙂

**Review by wirob — DISMISSED** [review]
Great job! I love it that you found the time to make some tests too! 🎊 🎉

**Review by kosmolet — APPROVED** [review]

**Review by kosmolet — DISMISSED** [review]

**Review by GayathriVaradan — APPROVED** [review]

**Review by zohresalimi — APPROVED** [review]

**Review by kosmolet — APPROVED** [review]

**Review by meerajm — APPROVED** [review]

**Pull request merged**